### PR TITLE
Add squash_to_root mount option

### PR DESCRIPTION
**Pull request by plopresti**
See issue https://github.com/containers/fuse-overlayfs/issues/241

### CODE REVIEW

File: main.c [hunk 1]
```diff
@@ -2688,7 +2697,10 @@ create_directory (struct ovl_data *lo, int dirfd, const char *name, const struct
   if (ret < 0)
     goto out;
 
-  if (uid != lo->uid || gid != lo->gid || get_upper_layer (lo)->has_stat_override || get_upper_layer (lo)->has_privileged_stat_override)
+  if (!lo->squash_to_root
```
Comment (giuseppe) [inline main.c:2700]: could we move this check inside do_fchown?

Comment (plopresti) [reply main.c]: Done.

Would similar handling make sense for do_chown and do_fchownat?

File: main.c [hunk 2]
```diff
@@ -218,6 +218,8 @@ static const struct fuse_opt ovl_opts[] = {
    offsetof (struct ovl_data, plugins), 0},
   {"xattr_permissions=%d",
    offsetof (struct ovl_data, xattr_permissions), 0},
+  {"squash_to_root",
```
Comment (giuseppe) [inline main.c:221]: should this be `squash_to_root=%d` or would it work as a boolean if we don't specify the type?

**Review by giuseppe — COMMENTED** [review]
one question, and could you please squash :-) the commits as well?

File: main.c (hunk 2 above)
Comment (plopresti) [reply main.c:221]: Seems to work as a Boolean. I was trying to follow the docs (https://libfuse.github.io/doxygen/structfuse__opt.html).

**Comment by plopresti** [comment]
> one question, and could you please squash :-) the commits as well?

Done. Also I think we do want the same handling for chown and fchownat, just for consistency. So I did that.

**Comment by plopresti** [comment]
@giuseppe Let me know whether this looks good or if there is something else I need to do. Thanks!

**Review by giuseppe — APPROVED** [review]
thanks, it looks good to me

**Comment by plopresti** [comment]
Thank you!

**Pull request merged**